### DISCUSSION TITLE
Welcome to MAGMATOMIX-UMD_package Discussions!

### DISCUSSION THREAD

**Discussion by rcaracas** [rcaracas/MAGMATOMIX-UMD_package#13] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by olivine-oliver** [comment]
Hi Razvan, I'm Oliver (working with Paul Asimow). I'm a big fan of the UMD scripts but I've come across a couple of issues with the LAMMPS to UMD script:
- LAMMPS numbers element types counting from 1, but UMD counts from 0, so there should be a little conversion (otherwise many other scripts crash, including the Speciation, Vibration, and MSD calculations)
- The elements of the pressure tensor are by default included in a log.lammps file and not in a separate pressure file 
- The true/unwrapped/diffusion-adjusted coordinates of atoms are indicated with "xu, yu, zu" in LAMMPS files, but the UMD script searches for "xa, ya, za" 
- The UMD files produced with this have a space in "Internal Energy" but later scripts do not (e.g. Full Averages)
- Broadly there is a lack of generalizability with the structure of the file, as it is not able to parse a series of run commands or reading from a log in which a restart file was used. A lot of LAMMPS runs are configured with multiple steps to equilibrate and/or initialize, or have changes in run conditions, so would be a benefit to making the script a bit more flexible. 

I also am running into an issue with the g(r)/ rdf script. The UMD script does not produce the same results as when I run the respective calculations with Ovito, LAMMPS, or other python scripts.

Again, I'm very grateful for these scripts - they've saved me a ton of time!

**Comment by rcaracas** [comment]
Hi Oliver,

Thanks for your email! I’m cc-ing this to Tim who, if I remember correctly, made the scripts, who might help. Isn’t it, Tim ? :) 

We’ll do the corrections and upload the new scripts asap.

Cheers,
Razvan


> On 16 Oct 2024, at 18:59, olivine-oliver ***@***.***> wrote:
> 
> 
> Hi Razvan, I'm Oliver (working with Paul Asimow). I'm a big fan of the UMD scripts but I've come across a couple of issues with the LAMMPS to UMD script:
> 
> LAMMPS numbers element types counting from 1, but UMD counts from 0, so there should be a little conversion (otherwise many other scripts crash, including the Speciation, Vibration, and MSD calculations)
> The elements of the pressure tensor are by default included in a log.lammps file and not in a separate pressure file
> The true/unwrapped/diffusion-adjusted coordinates of atoms are indicated with "xu, yu, zu" in LAMMPS files, but the UMD script searches for "xa, ya, za"
> The UMD files produced with this have a space in "Internal Energy" but later scripts do not (e.g. Full Averages)
> Broadly there is a lack of generalizability with the structure of the file, as it is not able to parse a series of run commands or reading from a log in which a restart file was used. A lot of LAMMPS runs are configured with multiple steps to equilibrate and/or initialize, or have changes in run conditions, so would be a benefit to making the script a bit more flexible.
> I also am running into an issue with the g(r)/ rdf script. The UMD script does not produce the same results as when I run the respective calculations with Ovito, LAMMPS, or other python scripts.
> 
> Again, I'm very grateful for these scripts - they've saved me a ton of time!
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/rcaracas/MAGMATOMIX-UMD_package/discussions/13#discussioncomment-10961849>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AMABC3ADSFYM7WQMGA2TJVDZ32ENRAVCNFSM6AAAAABPL3SR52VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAOJWGE4DIOI>.
> You are receiving this because you authored the thread.
> 

-- 
Dr. Razvan Caracas
Adjunct Professor
The Center for Earth Evolution and Dynamics (CEED)
University of Oslo
tel: +33-1-8395-7763
https://razvancaracas.info/
